### PR TITLE
Tony/mini contour downward

### PR DESCRIPTION
**Pull request by TonyLiu0226**
Implements mini contour downward algorithm
Tested it using the real life robot data with one crop row, and gives desired results in that situation.

### CODE REVIEW

File: algorithms/MiniContoursDownwards.py [hunk 1]
```diff
@@ -1,153 +1,188 @@
...
+        #intersections = Lines.get_intersections(lines)
+
+        #if not intersections:
+            #return frame, None
+
+        #x_points = [point[0] for point in intersections]
+        #y_points = [point[1] for point in intersections]
+
+        #calculate angle relative to a straight line downwards
+        #ref_line = cv2.line(frame, (0,0), (0, self.HEIGHT), self.color_1, 0)
```
Comment (L0Lmaker) [inline algorithms/MiniContoursDownwards.py]: What are these for?

File: algorithms/h [hunk 2]
```diff
@@ -0,0 +1,135 @@
+[[[483 253]
+  [331 253]
+  [530 266]
+  [459 266]
+  [365 266]
+  [469 279]
+  [376 279]
+  [500 292]
```
Comment (L0Lmaker) [inline algorithms/h]: Does this file needs to be removed?

File: config/algorithm/mini_contour_downward.yaml [hunk 3]
```diff
@@ -1,16 +1,12 @@
...
+high_green: [80, 255, 255]
 
-# cutOff parameters
-cut_off_factor: 0.9
-
-# parameters for Hough Lines Point Set
+# parameters for best fit line
 max_vote: 0
-num_strips: 30
+num_strips: 60
```
Comment (L0Lmaker) [inline config/algorithm/mini_contour_downward.yaml]: Could you add a description for what strips is meant for? For anyone that looks at the code with fresh eyes they might wonder what this is.

File: config/algorithm/mini_contour_downward.yaml [hunk 4]
```diff
@@ -1,16 +1,12 @@
...
 lines_max: 15
-threshold: 0
-min_rho: 0
-max_rho: 1000
-rho_step: 1
\ No newline at end of file
+param: 0
+reps: 0.01
+aeps: 0.01
```
Comment (L0Lmaker) [inline config/algorithm/mini_contour_downward.yaml]: These too if theres any info on it

File: algorithms/MiniContoursDownwards.py [hunk 5]
```diff
@@ -1,166 +1,187 @@
...
-        # other parameters for HoughLinesPointSet
-        # returns:
-        # frame: original frame with hough lines drawn on
-        # lines: list of [[votes, rho, theta]] of all lines
-        # point_lines: list of [votes, pt1, pt2] of all lines
-
+        # other parameters required to calculate line of best fit through centroids using cv2.fitLine
+        # returns: 
+        # frame: original frame with best fit line drawn on
+        # line: A vector [vx, vy, x, y] representing the line of best fit through all the centroids. [vx, vy] is a vector that describes the direction of the line, where (x, y) when taken together is a point on the line
```
Comment (L0Lmaker) [inline algorithms/MiniContoursDownwards.py]: Could you rewrite the function comments to be like this:
https://github.com/UBCAgroBot/jetson-ros-workspace/blob/b5887b2c2767b14363912719ebff574ed26666c7/src/helper_scripts/controller.py#L11

Much easier to read than with `#`s

**Comment by TonyLiu0226** [comment]
removed unneccesary config parameters in mini contour downwards

**Review by L0Lmaker — APPROVED** [review]

**Pull request merged**